spec/models/factory_spec.rb

```diff
@@ -81,6 +82,30 @@
         Factory.create_unique_factory(name: "factory1", traits: ["trait2_of_factory1"], assos:[])
         expect(Trait.any? { |trait| trait.name == "trait2_of_factory1" && trait.trait_relations.first.factory.name == "factory1" }).to eq true
       end
+
+      context  "create trait relation for already existing trait" do
+        let!(:existing_trait) { create :trait, name: "existing_trait" }
+        let!(:trait_relation) { create :trait_relation, factory_id: factory1.id, trait_id: existing_trait.id }
+        it "create new trait relation" do
+          expect{ Factory.create_unique_factory(name: "factory1", traits: ["existing_trait", "not_existing_trait"], assos:[]) }.to change { TraitRelation.all.size }.from(1).to(3)
+        end
+        it "don't create new trait" do
+          expect{ Factory.create_unique_factory(name: "factory1", traits: ["existing_trait", "not_existing_trait"], assos:[]) }.to change { Trait.all.size }.from(1).to(2)
```

Comment:
Metrics/LineLength: Line is too long. [170/160]<br>Style/SpaceAfterColon: Space missing after colon.<br>Style/SpaceBeforeBlockBraces: Space missing to the left of {.